s31175-PiotrTrybala/lab3-rozwiazanie.txt

```diff
@@ -0,0 +1,261 @@
+
+	Tak, powstał nowy plik
+
+Jaka jest treść względem poprzedniej próby bez programu tee?
+
+	Treść pliku zapisanego przy pomocy programu "tee" nie zawiera pliku,
+	w którym została zapisana
+
+Co robi program tee?
+
+	Program tee jednocześnie wypisuje do konsoli oraz zapisuje do pliku standardowe wyjście programu
+
+1 Zadania:
+
+A: Zobacz co się stanie jeśli wpiszemy w terminalu:
+
+ps
+
+	Wyświetlona została lista procesów
+
+ps -a
+
+	Wyświetlona została lista wszystkich procesów poza 2-ma procesami sesji
+
+ps x
+
+	Wyświetlona została lista wszystkich procesów aktualnego użytkownika
+
+ps axu
+INFO: tu kolejność parametrów a, x, u nie ma znaczenie a jedynie fakt, że są podane.
+Dowolna kolejność powinna dać ten sam wynik.
+
+	Wyświetlona została lista wszystkich procesów w formie "przyjaznej" dla użytkownika
+
+Jak myślisz, co oznacza znak zapytania w kolumnie 2?
+
+	Znak zapytanie w kolumnie "TTY" oznacza brak przypisanego do procesu terminala,
+	najprawdopodobniej są to procesy uruchomione na serwerze
+
+Nie wiesz? A gdzie może być wyjaśnienie?
+
+	man ps
+
+B: Jak wykonasz poniższe 2 zadania?
+
+Wyświetl wszystkie procesy bash
+
+	ps ax | grep bash
+
+Wyświetl wszystkie procesy należące do użytkownika root
+
+	ps -u root
+
+2 Zadanie:
+
+1.Uruchom wybrany przez Ciebie graficzny edytor tekstowy (np. gedit, gvim,
+Visual Studio Code, atom, itd...)
+
+	gedit
+
+2.Zobacz jaki ma on PID - przyda się do tego komenda ps
+
+	ps axu
+	PID procesu gedit to 2387
+
+3.Wydaj komendę kill w taki sposób, aby ten edytor się wyłączył. Zobacz czy to działa.
+
+	kill 2387
+	Zadziałało
+
+UWAGA: Niektóre programy przechwytują sygnały i mogą je częściowo blokować.
+Jeśli program nie wyłącza się, to zobacz jaka jest jego reakcja.
+Zobacz czy możesz wysłać do niego SIGKILL (gdzie sprawdzisz jak to zrobić?)
+
+	man kill
+	oraz
+	kill -l
+
+4.Zobacz działanie komendy killall bash
+
+	killall bash
+	Wynik to "operacja niedozwolona"
+
+5.Zobacz czy kill zadziała dla dowolnego procesu.
+
+	Nie mogę użyć komendy np na procesach innych użytkowników
+
+3
+
+1.Uruchom komendę cat bez parametrów
+
+	cat
+
+2.Wciśnij CTRL+C i zobacz co się stanie
+
+	Skrót CTRL-C wysyła sygnał przerwania programu
+
+Zobacz co się stanie:
+
+1.Wydaj komendę cat > wynik3_1.txt
+
+	cat > wynik3_1.txt
+
+2.Wpisz tekst witaj bez wciskania klawisza Enter
+
+	witaj
+
+3.Wciśnij CTRL+C
+
+4.Zobacz co się znalazło w pliku wynik3_1.txt
+
+	Plik jest pusty
+
+1.Wydaj komendę cat > wynik3_2.txt
+
+	cat > wynik3_2.txt
+
+2.Wpisz tekst witaj bez wciskania klawisza Enter
+
+	witaj
+
+3.Wciśnij CTRL+D (możliwe, że będzie trzeba wcisnąć go 2x)
+
+	Trzeba było
+
+4.Zobacz co się znalazło w pliku wynik3_2.txt
+
+	Plik zawiera tekst "witaj" bez znaku nowej linii
+
+W opisie rozwiązania zadania umieść:
+
+jak rozumiesz co się stało?
+
+	skrót CTRL-C przerwał działanie programu cat przed zapisaniem jego zawartości
+	skrót DTRL-D przerwał działanie po zapisaniu do pliku wprowadzonych danych
```

Comment:
Nie. CTRL+D nie przerywa działania programu.
Program otrzymał dzięki temu tylko pewną informację ale mimo wszystko pracował nadal.
A tak finalnit to nie `cat` zapisywał cokolwiek do pliku tylko bash przez przekierowanie stdout z cat do pliku.